src/mozanalysis/experiment.py

```diff
@@ -285,21 +294,131 @@ def get_per_client_data(
+    #     time_series_period='weekly', keep_client_id=False
+    # ):
+    #     time_limits = TimeLimits.for_ts(
+    #         self.start_date, self.num_dates_enrollment, last_date_full_data,
+    #         time_series_period
+    #     )
+    #     enrollments = self._process_enrollments(
+    #         enrollments, time_limits, time_series_period
+    #     )
+
+    #     # data_sources_and_metrics = process_metrics(metric_list)
+    #     data_sources_and_metrics = {
+    #         self._process_data_source(
+    #             data_source, time_limits
+    #         ).alias('data_source'): metric_list
+    #     }
+
+    #     res_per_ds = [
+    #         self._get_results_for_one_data_source(
+    #             enrollments, ds, ml, time_limits
+    #         )
+    #         for ds, ml in data_sources_and_metrics.items()
+    #     ]
+
+    #     res = reduce(lambda x, y: x.join(y, enrollments.columns), res_per_ds)
+
+    #     if not keep_client_id:
+    #         res = res.drop(enrollments.client_id)
+
+    #     res = res.drop(enrollments.analysis_window_end)
+    #     res = res.toPandas()
+
+    #     return {
+    #         t: res[res['analysis_window_start'] == t].drop('analysis_window_start')
+    #         for t in res['analysis_window_start'].unique()
+    #     }
+
+    def _get_results_for_one_data_source(
+        self, enrollments, data_source, metric_list, time_limits
+    ):
+        """Return a DataFrame of aggregated per-client metrics.
 
-        req_dates_of_data = analysis_start_days + analysis_length_days
+        Left join ``data_source`` to ``enrollments`` to get per-client
+        data within the ``time_limits``, then aggregate to compute the
+        requested metrics plus some sanity checks.
+        """
+        join_on = self._get_join_conditions(enrollments, data_source, time_limits)
 
-        enrollments = self.filter_enrollments_for_analysis_window(
-            enrollments, last_date_full_data, req_dates_of_data
-        ).alias('enrollments')
+        sanity_metrics = self._get_telemetry_sanity_check_metrics(
+            enrollments, data_source
+        )
 
-        data_source = self.filter_data_source_for_analysis_window(
-            data_source, last_date_full_data, analysis_start_days,
-            analysis_length_days
-        ).alias('data_source')
+        res = enrollments.join(
+            data_source,
+            join_on,
+            'left'
+        ).groupBy(
+            *[enrollments[c] for c in enrollments.columns]  # Yes, really.
+        ).agg(
+            *(metric_list + sanity_metrics)
+        )
+
+        return res
 
+    def _get_join_conditions(
+        self, enrollments, data_source, time_limits
+    ):
+        """Return a list of join conditions.
+
+        In ``_get_results_for_one_data_source``, we join ``enrollments``
+        to ``data_source``. This method returns the list of join
+        conditions.
+
+        The join conditions are:
```

Comment:
Is this documentation garbage? I'm just restating the code and the comments that are already in the code... Shall I just remove it?